SmartTutorTests/Unit/SubmissionTests.cs

```diff
@@ -68,22 +99,24 @@ public void Evaluates_answer_submission(QuestionSubmission submission, List<bool
             correctness.ShouldBe(expectedCorrectness);
         }
 
+        private static readonly Submission Submission = new Submission(1, 1, true, DateTime.Now);
```

Comment:
Remove this. Submission should be an abstract class as it does not make sense as a standalone object (submission of what?). As a side note, we should avoid sharing state between tests, even when it's immutable - if this test suite grows and more tests start to depend on this shared field, what happens when one test changes it?

Why do you need these fields in the test data? If it turns out we don't, don't forget to remove the additional parameter you added to the QuestionSubmission constructor, and the two constructors you added to Submission.